devops/pypi-release-new.yml

```diff
@@ -0,0 +1,257 @@
+      kvPassword: passwordtest
+      pypiUrl: https://test.pypi.org/legacy/
+    ${{ if eq(parameters.releaseType, 'Production')}}:
+      kvUsername: 'usernameprod'
+      kvPassword: 'passwordprod'
+      pypiUrl: https://upload.pypi.org/legacy/
+
+  jobs:
+  - deployment: 'PyPI_${{parameters.releaseType}}_Upload'
+    displayName: PyPI ${{parameters.releaseType}} Upload
+    ${{ if eq(parameters.releaseType, 'Test')}}:
+      environment: 'PyPI-Test Deployment'
+    ${{ if eq(parameters.releaseType, 'Production')}}:
+      environment: 'PyPI Deployment'
+    pool:
+      vmImage: $(poolImage)
+
+    strategy:
+      runOnce:
+        deploy:
+          steps:
+          - task: UsePythonVersion@0
+            displayName: 'Use Python $(poolPythonVersion)'
+            inputs:
+              versionSpec: $(poolPythonVersion)
+              addToPath: true
+          
+          - script: pip install twine
+            displayName: 'Install twine'
+
+          - task: AzureKeyVault@1
+            displayName: "Fetch deployment secrets from KeyVault"
+            inputs:
+              azureSubscription: ${{variables.kvSubscription}}
+              keyVaultName:  ${{variables.kvVaultName}}
+              secretsFilter: '${{variables.kvUsername}},${{variables.kvPassword}}'
+
+          # Quote the password argument in case it contains characters the shell might try to parse
+          - script: 'twine upload --verbose --repository-url $(pypiUrl) -u $(${{variables.kvUsername}}) -p "$(${{variables.kvPassword}})" $(Pipeline.Workspace)/$(packageArtifactName)/*'
+            displayName: Upload to ${{parameters.releaseType}} PyPI
+
+        # TODO: Add GitHub Release task, so links in PyPI ReadMe will work without manual intervention
+  
+  - job: PyPI_Pause
+    pool: server
+    dependsOn: 'PyPI_${{parameters.releaseType}}_Upload'
+    displayName: PyPI Pause
+
+    steps:
+    - task: Delay@1
+      displayName: "Pause to allow PyPI updates to complete"
+      inputs:
+        delayForMinutes: "5"
+
+# # ==================================================================================================
+
+- stage: TestFromPyPI
+  displayName: Test package from ${{parameters.releaseType}} PyPI
+  dependsOn: UploadWheel
+  pool:
+    vmImage: $(poolImage)
+  
+  variables:
+    freezeArtifactStem: TestPyPIFreeze
+    freezeFileStem: requirements-pypi-test
+
+  jobs:
+  - template: templates/all-tests-job-template.yml
+    parameters:
+      platforms:  { Linux: ubuntu-latest, Windows: vs2017-win2016 }
+      pyVersions: [3.5, 3.6, 3.7, 3.8]
+      freezeArtifactStem: $(freezeArtifactStem)
+      freezeFileStem: $(freezeFileStem)
+      installationType: 'PyPI'
+      targetType: ${{parameters.releaseType}}
+      versionArtifactName: $(versionArtifactName)
+      versionArtifactFile: $(versionFileName)
+
+  - template: templates/all-tests-job-template.yml
+    parameters:
+      platforms: { MacOS: macos-latest }
+      testRunTypes: ['Unit']
```

Comment:
This one has Unit only, while Windows/Linux above doesn't have that. Why is that?

Reply:
We've never run notebook tests of MacOS. We should probably look into that again, but that's not for this PR.